quantum_enablement/circuits/_mbl.py

```diff
@@ -0,0 +1,123 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Many-body localization (MBL) quantum circuits.
+
+[1] Shtanko et.al. Uncovering Local Integrability in Quantum Many-Body Dynamics,
+    https://arxiv.org/abs/2307.07552
+"""
+
+
+from numpy import pi
+from qiskit.circuit import Parameter, ParameterVector, QuantumCircuit
+
+
+class MBLChainCircuit(QuantumCircuit):
+    """Parameterized MBL non-periodic chain (i.e. 1D) quantum circuit.
+
+    Parameters correspond to interaction strength (θ), and
+    disorders vector (φ) with one entry per qubit. In 1D,
+    θ < 0.16π ≈ 0.5 corresponds to the MBL regime; beyond such
+    critical value the dynamics become ergodic (i.e. thermal).
+    Disorders are random on a qubit by qubit basis [1].
+
+    Args:
+        num_qubits: number of qubits (must be even).
+        depth: two-qubit depth.
```

Comment:
I saw the depth validation requires depth to be even, it will be good to also document this in the description of the args